redis/src/caching/config.rs

```diff
@@ -0,0 +1,49 @@
+use std::num::NonZeroUsize;
+use std::time::Duration;
+
+/// Defines CacheMode of CacheConfig
+#[derive(Clone, PartialEq, Debug, Copy)]
+pub enum CacheMode {
+    /// Don't cache
+    None,
```

Comment:
why is there a `None` mode? wouldn't a disabled cache just be a `let cache_config: Option<CacheMode> = None;`?

Reply:
Hmm I will refactor CacheConfig part in general, that might be better